types/catbox/index.d.ts

```diff
@@ -21,36 +18,39 @@ export type CallBackWithResult<T> = (err: Boom.BoomError | null | undefined, res
-     *  * key - a cache key object (see [ICacheKey]).
-     *  * callback - a function with the signature function(err, cached). If the item is not found, both err and cached are null. If found, the cached object is returned
-     */
-    get(key: CacheKey, callback: CallBackWithResult<null | CachedObject>): CacheItem;
-    /**
-     * set(key, value, ttl, callback) - store an item in the cache for a specified length of time, where:
-     *  * key - a cache key object (see [ICacheKey]).
-     *  * value - the string or object value to be stored.
-     *  * ttl - a time-to-live value in milliseconds after which the item is automatically removed from the cache (or is marked invalid).
-     *  * callback - a function with the signature function(err).
-     */
-    set(key: CacheKey, value: CacheItem, ttl: number, callback: CallBackNoResult): void;
-    /**
-     * drop(key, callback) - remove an item from cache where:
-     *  * key - a cache key object (see [ICacheKey]).
-     *  * callback - a function with the signature function(err).
-     */
-    drop(key: CacheKey, callback: CallBackNoResult): void;
-    /** isReady() - returns true if cache engine determines itself as ready, false if it is not ready. */
-    isReady(): boolean;
-    /** validateSegmentName(segment) - returns null if the segment name is valid (see below), otherwise should return an instance of Error with an appropriate message. */
-    validateSegmentName(segment: string): null | Boom.BoomError;
+  constructor(engine: EnginePrototypeOrObject, options: ClientOptions);
+
+  /** start() - creates a connection to the cache server. Must be called before any other method is available. */
+  start(): Promise<void>;
+
+  /** stop() - terminates the connection to the cache server. */
+  stop(): void;
+
+  /**
+   * get(key, callback) - retrieve an item from the cache engine if found where:
+   *  * key - a cache key object (see [ICacheKey]).
+   */
+  get(key: CacheKey): Promise<null | CachedObject>;
+
+  /**
+   * set(key, value, ttl, callback) - store an item in the cache for a specified length of time, where:
+   *  * key - a cache key object (see [ICacheKey]).
+   *  * value - the string or object value to be stored.
+   *  * ttl - a time-to-live value in milliseconds after which the item is automatically removed from the cache (or is marked invalid).
+   */
+  set(key: CacheKey, value: CacheItem, ttl: number): Promise<void>;
+
+  /**
+   * drop(key, callback) - remove an item from cache where:
+   *  * key - a cache key object (see [ICacheKey]).
+   */
+  drop(key: CacheKey): Promise<void>;
+
+  /** isReady() - returns true if cache engine determines itself as ready, false if it is not ready. */
+  isReady(): boolean;
+
+  /** validateSegmentName(segment) - returns null if the segment name is valid (see below), otherwise should return an instance of Error with an appropriate message. */
+  validateSegmentName(segment: string): null | Error;
```

Comment:
Why change it to `Error` instead of `BoomError`?

Reply:
The API makes no mention of `Boom` errors, and even if the code suggests they will all be `Boom` instances, I always tend to go with the API reference:

https://github.com/hapijs/catbox

Also, as of Boom v5, there's no `BoomError` anymore, the equivalent would be `Boom.Boom`.

Reply:
(and Boom.Boom extends Error)

Reply:
Yea that's good enough for me.